src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -115,33 +115,112 @@ public static void Analyze(CSharpCompilation compilation, Symbol member, BoundNo
 
         protected override void Normalize(ref LocalState state)
         {
+            int oldNext = state.Capacity;
             state.EnsureCapacity(nextVariableSlot);
+            for (int slot = oldNext; slot < nextVariableSlot; slot++)
+            {
+                var value = GetDefaultState(ref state, slot);
+                state[slot] = value;
+            }
+        }
+
+        private bool? GetDefaultState(ref LocalState state, int slot)
+        {
+            if (slot == 0)
+            {
+                return null;
+            }
+
+            var variable = variableBySlot[slot];
+            var symbol = variable.Symbol;
+
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Local:
+                    return null;
```

Comment:
>null [](start = 27, length = 4)

I don't understand why we want Normalize to set all local variable states to null. What is Normalize used for?

Reply:
Oh, I see, when expanding the state vector.

---
In reply to: [154791949](https://github.com/dotnet/roslyn/pull/23063#discussion_r154791949) [](ancestors = 154791949)